Python: Initial Bedrock support

mindsolve opened the pull request:
## Proposed Changes
- Adds initial support for Bedrock servers (unconnected ping packets), see #45
- Adds type annotations, where necessary or useful
- Bumps Python package version to 2.3.0
- Adds full documentation of all attributes

@ldilley: Could you have a look at the below questions/problems? What do you think? And should this be done before merging this PR? Thanks!

## Known problems/improvement potential
- `bedrock_raknet_query` currently behaves "TCP-like", assuming no dropped packets.
  - A reliability should be implemented (resending of packets)
- Many places still speak of SLP protocol, even though they mean/contain all possible protocols.
  - Should be renamed to something general, such as simply `Protocol`
  - Serverquery could be implemented in a similar way, as just another request protocol.
- ~The `extra_data` object is only used in the bedrock protocol~
  - ~Should be used by all protocols exposing additional data, that is not covered by the common properties.~

## Other open points
- [x] Whether to include the MotD line 2 in the `motd` property (as second line, like in the other protocols), or include it in the version property (like in the ruby edition)
- [x] Whether to include the `extra_data` property at all, or add properties that are only used for single protocols.
  - Decided to not use the `extra_data` property, instead to add protocol specific attributes

ldilley commented:
This looks good, @mindsolve. Thank you for taking it on.

Unfortunately, I am unable to test this code currently since it seems that the Python version on my development system is too old (the latest version in the Debian repository on this system is 3.5.3). Support for the following syntax must have come later considering that version 3.5.3 chokes on it:

```python
142     self.address: str = address
143     self.port: int = port
144     self.online: bool = False
```

```python
<22:50:12> (~/projects/tmp/minestat/Python) [:a745e7]
[ldilley@windev] :) {3372}$ python3 example.py
Traceback (most recent call last):
  File "example.py", line 1, in <module>
    import minestat
  File "/home/ldilley/projects/tmp/minestat/Python/minestat/__init__.py", line 142
    self.address: str = address
                ^
SyntaxError: invalid syntax
```

I'll work on getting a Python >=3.8.x interpreter installed to test this PR.

Do not worry about the failing Travis CI build for Java. It is an issue with OpenJDK 9 which I've addressed in PR #98.

In reference to your other comments/questions:
1. We can probably revisit making the UDP implementation more reliable later if necessary. I would not worry about it now (unless you really want to).
2. Does the Ruby implementation from PR #98 look OK to you as it pertains to "SLP" and "protocol" use? I attempted to denote UDP/Bedrock/MCPE usage where applicable (in constant names, comments, etc.). Perhaps we could use this as a guide and make certain areas of the Python code more generic/specific wherever necessary.
3. If you want to make `extra_data` available for other information from JSON queries (for example), I am fine with that. I did something similar with the game mode field (`@mode`) only being populated after Bedrock queries in the Ruby edition.
4. For consistency between language implementations and per our Discord discussion, I'd opt for including the MotD 2 value in the version string. We can always modify this behavior if you feel a change is appropriate of course.
5. I'd say keep `extra_data` since some end users might find the additional information useful.

ldilley commented:
I've compiled the latest Python (3.10.4) and was able to successfully test the code from this PR. It seems to work as expected per the output below.

```python
<12:52:45> (~/projects/tmp/minestat/Python) [:a745e7 *]
[ldilley@windev] :) {3430}$ python3 example.py
Minecraft server status of test.frag.land on port 19132:
Server is online running version 1.18.30 with 0 out of 32 players.
Message of the day: §cTest Server
PocketMine-MP
Message of the day without formatting: Test Server
PocketMine-MP
Latency: 19ms
Connected using protocol: BEDROCK_RAKNET
```

I just have 2 suggestions prior to merging based on our previous discussion:
1. Can we see about including MotD 2 along with edition in the version string? MotD 2 seems out of place on a line by itself.
2. Can we update the example to include the game mode (creative, survival, etc.) if protocol is `BEDROCK_RAKNET`?

Please let me know what you think. Excellent job too, @mindsolve. :)

mindsolve commented:
Hi @ldilley!

Thanks a lot for the review and the compliments :)
I have removed the `extra_data` attribute for now and added an attribute for the Bedrock gamemode.
Additionally, I have added a list of all supported attributes to the Python specific README.md.

1.: I have gone ahead and changed the `version` to the same format as the Ruby implementation.
2.: The example in both the Python README and the "global" one now contain a reference to the Bedrock gamemode.
